### PR TITLE
try except with pathlib and pathlib2

### PR DESCRIPTION
**Pull request by lepmik**

### CODE REVIEW

**Comment by lepmik** [comment]
https://github.com/CINPLA/exdir/issues/74

File: exdir/core/exdir_file.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import os
 import shutil
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline exdir/core/exdir_file.py:6]: No need to wrap the second attempt in a try-block

File: exdir/core/exdir_file.py [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+        import pathlib2 as pathlib
+    except ImportError:
```
Comment (dragly) [inline exdir/core/exdir_file.py:8]: You can remove this as well (but if you don't, it is missing an `as e`).

File: exdir/core/exdir_object.py [hunk 3]
```diff
@@ -5,7 +5,13 @@
 from enum import Enum
 import os
 import warnings
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline exdir/core/exdir_object.py:11]: Ditto

File: exdir/core/group.py [hunk 4]
```diff
@@ -1,6 +1,12 @@
 import os
 import re
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline exdir/core/group.py:6]: Ditto

File: exdir/core/validation.py [hunk 5]
```diff
@@ -1,6 +1,12 @@
 from enum import Enum
 import os
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline exdir/core/validation.py:6]: Ditto

File: exdir/plugins/git_lfs.py [hunk 6]
```diff
@@ -1,6 +1,12 @@
 import subprocess
 import exdir.plugin_interface
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline exdir/plugins/git_lfs.py:6]: Ditto

File: exdir/utils/display.py [hunk 7]
```diff
@@ -1,4 +1,10 @@
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline exdir/utils/display.py:4]: Ditto

File: exdir/utils/path.py [hunk 8]
```diff
@@ -1,4 +1,10 @@
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline exdir/utils/path.py:4]: Ditto

File: tests/conftest.py [hunk 9]
```diff
@@ -4,7 +4,13 @@
 import shutil
 import os
 import h5py
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline tests/conftest.py:10]: Ditto

File: tests/test_file.py [hunk 10]
```diff
@@ -14,7 +14,13 @@
 
 import pytest
 import os
-import pathlib
+try:
+    import pathlib
+except ImportError as e:
+    try:
```
Comment (dragly) [inline tests/test_file.py:20]: Ditto, and, I guess you get the point. The following do not need the try either.

**Review by dragly — CHANGES_REQUESTED** [review]

**Comment by lepmik** [comment]
The reason I put the second try except is to raise the error from the first try except . That is, if you don't have either pathlib or pathlib2 you will get an error saying you don't have pathlib, not pathlib2 as your suggestion would result in.

**Comment by lepmik** [comment]
@dragly do you concur?

**Review by dragly — APPROVED** [review]
Sorry, did not see your reply before now. Yes, then I agree. Nice trick!

**Pull request merged**